tests/unit/net/data_loader.rs

```diff
@@ -69,8 +72,23 @@ fn plain_charset() {
 }
 
 #[test]
+fn plain_only_charset() {
+    assert_parse(
+        "data:charset=utf-8,hello",
```

Comment:
The correct shorthand is `data:;charset=utf-8,hello`.  `data:charset=utf-8,<b>hello` gets interpreted as HTML by Firefox.